### PR TITLE
Tech/services refactoring

### PR DESCRIPTION
**Pull request by vladilyushchenko**

### CODE REVIEW

File: src/main/java/com/fbc/bot/service/MessageTypeService.java [hunk 1]
```diff
@@ -1,9 +1,35 @@
...
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class MessageTypeService {
+
+    private final MessageTypeMatcher typeMatcher;
+    private final Set<MessageType> inlineMessageTypes = Set.of(INLINE_SHARE_COCK_SIZE, INLINE_UNKNOWN);
+
+    // in future take into account current user's bot state for interview-chatting
+    public MessageType getUserMessageType(Update update) {
```
Comment (zaplitny) [inline src/main/java/com/fbc/bot/service/MessageTypeService.java:23]: Do we really need separate service class for it?

Comment (vladilyushchenko) [reply src/main/java/com/fbc/bot/service/MessageTypeService.java:23]: In future I want to add conversation with bot like answering questions for making profiles. For this purpose I need to store user's state and accordingly to this state choose MessageType for the next answer

File: src/main/java/com/fbc/bot/telegram/handler/MessageHandlerFactory.java [hunk 2]
```diff
@@ -16,7 +16,7 @@ public class MessageHandlerFactory {
     private final Map<MessageType, MessageHandler> handlers;
 
     public MessageHandlerFactory(List<MessageHandler> handlers) {
```
Comment (zaplitny) [inline src/main/java/com/fbc/bot/telegram/handler/MessageHandlerFactory.java:18]: It's not a factory. Factory creates objects. Take a look at https://stackoverflow.com/questions/20918825/can-spring-autowired-map. Also previous version could be rewritten to 
`this.handlers = handlers.stream().collect(
                Collectors.toMap(MessageHandler::getMessageType, Function.identity())
        )`

File: src/main/java/com/fbc/bot/service/telegram/CockSizeServiceFacade.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+@Component
+@RequiredArgsConstructor
+public class CockSizeServiceFacade {
+
+    private final CockSizeService cockSizeService;
+    private final UserService userService;
+
+    @Transactional
+    public String getCockSizeAnswer(org.telegram.telegrambots.meta.api.objects.User tgUser) {
```
Comment (zaplitny) [inline src/main/java/com/fbc/bot/service/telegram/CockSizeServiceFacade.java:19]: Add tests for new class

File: src/main/java/com/fbc/bot/service/UserService.java [hunk 4]
```diff
@@ -1,19 +1,46 @@
...
+        User user = mapper.toEntity(tgUser);
+        user.setUserStatus(UNKNOWN);
+        return repository.save(user);
+    }
+
+    public UserDto updateUser(User user) {
+        return mapper.toDto(repository.save(user));
+    }
+
+    public UserDto getUserById(Long id) {
```
Comment (zaplitny) [inline src/main/java/com/fbc/bot/service/UserService.java:36]: @Transactional(readOnly=true)

File: src/test/java/com/fbc/bot/service/impl/CockSizeServiceTest.java [hunk 5]
```diff
@@ -2,6 +2,7 @@
```
Comment (zaplitny) [inline src/test/java/com/fbc/bot/service/impl/CockSizeServiceTest.java:2]: you removed impl package

**Review by zaplitny — APPROVED** [review]

**Pull request merged**